### DISCUSSION TITLE
Naman Goel - Stylex

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#307] [Announcements]
- https://x.com/naman34
- https://x.com/stylexjs
- https://stylexjs.com/

### StyleX

- [x] What is stylex? How is it different than other css in js solutions?
- [x] Is it just for react?
- [x] How long has it existed at facebook? Were there other frameworks that competed and died off?
- [x] How does it work under the hood? What type of code does it deliver to the browser?
- [x] At runtime what is the cost of running it? What situations get me to a higher runtime cost?
- [ ] ~Does it support typescript? What can I enforce?~
- [x] Does it put any limiations on the CSS I can write?
- [x] It relies on a babel plugin to do it's thing. How does that work at facebook scale? In particular I could see building time becoming a problem with babel in the toolchain.
- [x] Are there any plans to support other build tools?
- [x] How does stlyex garuntee that styles that are deterministic? Ive been bitten a few times by import order changing in the final product and breaking some styling.

### Programming

- [x] What do you think is the programming language we should all be using instead of javascript?
- [x] Facebook has a lot of it's own languages (Flow, hack). Why do you think those haven't taken off? 
- [x] Why is flow better than typescript?

### Future

- [ ] What's planned for stylex?
- [ ] What do you think the future of styling is?

**Comment by nmn** [comment]
> Facebook has a lot of it's own languages (flex, hack)

Did you mean `Flow` and Hack?
